nav2_motion_primitives/src/back_up.cpp

```diff
@@ -81,16 +90,74 @@ nav2_tasks::TaskStatus BackUp::controlledBackup()
   double diff_y = initial_pose_->pose.pose.position.y - current_odom_pose->pose.pose.position.y;
   double distance = sqrt(diff_x * diff_x + diff_y * diff_y);
 
-  if (distance >= abs(command_x_)) {
+  if (distance >= std::abs(command_x_)) {
     cmd_vel.linear.x = 0;
     robot_->sendVelocity(cmd_vel);
+    RCLCPP_INFO(node_->get_logger(), "Completed backup, traveled %0.2f", distance);
     return TaskStatus::SUCCEEDED;
   }
+
   // TODO(mhpanah): cmd_vel value should be passed as a parameter
-  command_x_ < 0 ? cmd_vel.linear.x = -0.025 : cmd_vel.linear.x = 0.025;
+  if (command_x_ < 0) {
+    cmd_vel.linear.x = -1.0 * default_vel_.linear.x;
+  } else {
+    cmd_vel.linear.x = default_vel_.linear.x;
+  }
+
   robot_->sendVelocity(cmd_vel);
 
   return TaskStatus::RUNNING;
 }
 
+bool BackUp::pathIsClear()
+{
+  auto robot_pose = std::make_shared<geometry_msgs::msg::PoseWithCovarianceStamped>();
+
+  nav2_world_model::FreeSpaceServiceRequest request;
+
+  // Check for clear path relative to robot.
+  request.frame_id = "base_link";
+  request.reference.x = 0.0;
+  request.reference.y = 0.0;
+  request.rotation = 0.0;
+
+  // TODO(orduno) If robot is localized we might want to check relative to the map
```

Comment:
Commented out code. Remove if not needed.

Reply:
removed